starlette/testclient.py

```diff
@@ -426,30 +429,30 @@ def request(  # type: ignore[override]
         method: str,
         url: httpx._types.URLTypes,
         *,
-        content: httpx._types.RequestContent = None,
-        data: httpx._types.RequestData = None,
-        files: httpx._types.RequestFiles = None,
+        content: typing.Optional[httpx._types.RequestContent] = None,
+        data: typing.Optional[_RequestData] = None,
+        files: typing.Optional[httpx._types.RequestFiles] = None,
         json: typing.Any = None,
-        params: httpx._types.QueryParamTypes = None,
-        headers: httpx._types.HeaderTypes = None,
-        cookies: httpx._types.CookieTypes = None,
+        params: typing.Optional[httpx._types.QueryParamTypes] = None,
+        headers: typing.Optional[httpx._types.HeaderTypes] = None,
+        cookies: typing.Optional[httpx._types.CookieTypes] = None,
         auth: typing.Union[
             httpx._types.AuthTypes, httpx._client.UseClientDefault
         ] = httpx._client.USE_CLIENT_DEFAULT,
-        follow_redirects: bool = None,
-        allow_redirects: bool = None,
+        follow_redirects: typing.Optional[bool] = None,
+        allow_redirects: typing.Optional[bool] = None,
         timeout: typing.Union[
             httpx._client.TimeoutTypes, httpx._client.UseClientDefault
         ] = httpx._client.USE_CLIENT_DEFAULT,
-        extensions: dict = None,
+        extensions: typing.Optional[typing.Dict[str, typing.Any]] = None,
     ) -> httpx.Response:
         url = self.base_url.join(url)
         redirect = self._choose_redirect_arg(follow_redirects, allow_redirects)
         return super().request(
             method,
             url,
             content=content,
-            data=data,
+            data=data,  # type: ignore
```

Comment:
Please add the code error :pray:

Reply:
I didn't get it. The reason is because httpx defines its own type?

Reply:
The reason is that httpx's current type is `dict` and we'll be giving it a `Mapping`. The correct type is `Mapping` and it will be fixed in the next httpx release.

Reply:
Ah! Got it. :+1:

Reply:
Ok, another question then... Why are you being more strict here than `httpx`? Or the intention is to also be more strict there?

Reply:
I didn't even pay attention to that to be honest. I would say yes, we should be more strict there. I can make a PR for that. Happy to change this to `Mapping[str, Any]` in the meantime, but I think it's better to be strict and then relax than the other way around (to avoid breaking changes)